### DISCUSSION TITLE
Github action matrix strategy how to set max-parallel dynamically

### DISCUSSION THREAD

**Discussion by rajammavenkatesh** [community/community#141124] [Actions] [answered]
### Select Topic Area

Question

### Body

I need to set max-parallel dynamically, Based input ex: If deploy to integration I need to set 1 else 4

**Comment by VeerammalV** [comment, marked as answer]
To set `max-parallel` dynamically based on input (e.g., integration environment vs others), you can use an environment variable or input parameter to control the value. Since GitHub Actions doesn't directly support dynamic `max-parallel` values within a single matrix job, you can achieve it by splitting into separate jobs with different configurations based on the input.

Here's a more flexible approach using conditional jobs:

jobs:
  deploy:
    strategy:
      matrix:
        env: [integration, production, staging]
    max-parallel: ${{ github.event.inputs.env == 'integration' && 1 || 4 }}
    steps:
      - name: Deploy to ${{ matrix.env }}
        run: echo "Deploying to ${{ matrix.env }}"

This uses the input to set `max-parallel` dynamically based on the environment. If `env` is 'integration', it will set `max-parallel` to 1; otherwise, it will set it to 4.

**Reply by rajammavenkatesh** [reply]
Thank you @VeerammalV
